src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -121,6 +121,30 @@ const DiffExprCompute = (props) => {
     );
   };
 
+  const renderError = (type, description) => (
+    <Alert
+      message={type.charAt(0).toUpperCase() + type.slice(1)}
+      type={type}
+      showIcon
+      description={description}
+    />
+  );
+  const checkAndRenderError = () => {
+    const canRun = canRunDiffExpr();
```

Comment:
ok, it is looking better, thanks for making the changes. But I don't think we need separate `renderError` and `checkAndRenderError` functions, because it is a bit confusing. You can have 1 function that:

if there is no error, just returns what you have in the bottom of `checkAndRenderError` (this `<Form size='smal'`, ... etc) and it there is an error:

configures the value of `description` and then returns
```
 <Alert
      message={type.charAt(0).toUpperCase() + type.slice(1)}
      type={type}
      showIcon
      description={description}
    />
```
if there was an error and before